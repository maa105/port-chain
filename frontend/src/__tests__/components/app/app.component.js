import React from 'react';
import { render } from '@testing-library/react';

import AppComponent from '../../../components/app/app.component';
import LayoutComponent from '../../../components/layout/layout.component';
import { setState } from 'react-redux';

jest.mock('../../../components/layout/layout.component', () => {
  return jest.fn(() => <div>Layout Mock</div>)
});
jest.mock('../../../components/vessels/vessels.component.js', () => {
  return jest.fn(() => <div>Vessels Page</div>)
});
jest.mock('../../../components/ports/ports.component.js', () => {
  return jest.fn(() => <div>Ports Page</div>)
});

jest.mock('react-redux', () => {
  var state = {};
  return {
    setState: (_state) => { state = _state; },
    useSelector: jest.fn((f) => f(state))
  }
});

beforeEach(() => {
  jest.clearAllMocks();
});

it('renders nulls for invalid page', () => {
  setState({ router: { routerState: { page: 'page' } } });
  const component = render(
    <AppComponent />
  );
  expect(component.asFragment()).toMatchSnapshot();
  
  expect(LayoutComponent.mock.calls.length).toEqual(1);
  
  const [VesselsPage, PortsPage] = LayoutComponent.mock.calls[0][0].children;

  expect(VesselsPage).toEqual(null);
  expect(PortsPage).toEqual(null);
});

it('renders vessels page', () => {
  setState({ router: { routerState: { page: 'vessels' } } });
  const component = render(
    <AppComponent />
  );
  expect(component.asFragment()).toMatchSnapshot();
  
  expect(LayoutComponent.mock.calls.length).toEqual(1);
  
  const [VesselsPage, PortsPage] = LayoutComponent.mock.calls[0][0].children;

  expect(VesselsPage).not.toEqual(null);
  expect(PortsPage).toEqual(null);

  const vesselsComponent = render(VesselsPage);
  expect(vesselsComponent.asFragment()).toMatchSnapshot();
});

it('renders ports page', () => {
  setState({ router: { routerState: { page: 'ports' } } });
  const component = render(
    <AppComponent />
  );
  expect(component.asFragment()).toMatchSnapshot();
  
  expect(LayoutComponent.mock.calls.length).toEqual(1);
  
  const [VesselsPage, PortsPage] = LayoutComponent.mock.calls[0][0].children;

  expect(VesselsPage).toEqual(null);
  expect(PortsPage).not.toEqual(null);

  const portsComponent = render(PortsPage);
  expect(portsComponent.asFragment()).toMatchSnapshot();
});
