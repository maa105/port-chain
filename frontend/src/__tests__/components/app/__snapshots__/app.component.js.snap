// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nulls for invalid page 1`] = `
<DocumentFragment>
  <div>
    Layout Mock
  </div>
</DocumentFragment>
`;

exports[`renders ports page 1`] = `
<DocumentFragment>
  <div>
    Layout Mock
  </div>
</DocumentFragment>
`;

exports[`renders ports page 2`] = `
<DocumentFragment>
  <div>
    Ports Page
  </div>
</DocumentFragment>
`;

exports[`renders vessels page 1`] = `
<DocumentFragment>
  <div>
    Layout Mock
  </div>
</DocumentFragment>
`;

exports[`renders vessels page 2`] = `
<DocumentFragment>
  <div>
    Vessels Page
  </div>
</DocumentFragment>
`;
